### PR TITLE
Add a nightly production backup that runs at 03:30

### PR DESCRIPTION
**Pull request by peteryates**
### Context

The process is copied from ECF's. It uses pg_dump and then uploads the resulting file to a storage account. Unlike ECF's which runs at 03:00 this runs at 03:30.

### CODE REVIEW

File: .github/actions/backup-and-upload-database/action.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+    - name: Backup database
+      shell: bash
+      run: |
+        bin/konduit.sh npq-registration-production-web -- pg_dump -E utf8 --compress=1 --clean --if-exists --no-privileges --no-owner --verbose -f backup-production.sql.gz
+
+    - name: Set connection string
+      shell: bash
+      run: |
+        STORAGE_CONN_STR=$(az storage account show-connection-string -g s189p01-cpdnpq-pd-rg -n s189p01cpdnpqdbbkppdsa --query 'connectionString')
```
Comment (peteryates) [inline .github/actions/backup-and-upload-database/action.yml:40]: The storage account `s189p01cpdnpqdbbkppdsa` doesn't exist yet. @cwrw, did you just create this manually using Azure Portal in ECF?

File: .github/actions/backup-and-upload-database/action.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        secrets: "SLACK-WEBHOOK"
+
+    - name: Notify Slack channel on job failure
+      if: failure()
+      uses: rtCamp/action-slack-notify@v2
+      env:
+        SLACK_USERNAME: CI Deployment
+        SLACK_TITLE: Database backup failure
+        SLACK_MESSAGE: Production database backup job failed
+        SLACK_WEBHOOK: ${{ steps.key-vault-secrets.outputs.SLACK-WEBHOOK }}
```
Comment (peteryates) [inline .github/actions/backup-and-upload-database/action.yml:68]: We don't have a `SLACK-WEBHOOK` yet, I've put in a request for one and will add it once received.

File: .github/actions/backup-and-upload-database/action.yml (hunk 1 above)
Comment (saliceti) [reply .github/actions/backup-and-upload-database/action.yml:40]: It should be created automatically by https://github.com/DFE-Digital/npq-registration/blob/main/terraform/application/database.tf#L13

**Review by saliceti — APPROVED** [review]

File: .github/actions/backup-and-upload-database/action.yml (hunk 1 above)
Comment (cwrw) [reply .github/actions/backup-and-upload-database/action.yml:40]: when we checked however last time in ECF it already existed so this should exist already for NPQ?

Comment (cwrw) [reply .github/actions/backup-and-upload-database/action.yml:40]: act nvm I checked and `s189p01cpdnpqdbbkppdsa` already exists so all good

File: .github/actions/backup-and-upload-database/action.yml (hunk 2 above)
Comment (cwrw) [reply .github/actions/backup-and-upload-database/action.yml:68]: maybe can be to the same shared alerts channel for NPQ/ECF?
as this is set to both now, our ECF specific one and the shared one?

File: .github/actions/backup-and-upload-database/action.yml (hunk 1 above)
Comment (peteryates) [reply .github/actions/backup-and-upload-database/action.yml:40]: Ah, thanks. I looked around the code and couldn't see any reference to it being built!

File: .github/actions/backup-and-upload-database/action.yml (hunk 2 above)
Comment (peteryates) [reply .github/actions/backup-and-upload-database/action.yml:68]: Good call, reusing the ECF prod one for now 👌🏽

**Pull request merged**